pythonforandroid/recipes/python3/__init__.py

```diff
@@ -71,17 +70,22 @@ class Python3Recipe(TargetPythonRecipe):
         # Python 3.8.1 & 3.9.X
         ('patches/py3.8.1.patch', version_starts_with("3.8")),
         ('patches/py3.8.1.patch', version_starts_with("3.9")),
-        ('patches/py3.8.1.patch', version_starts_with("3.10"))
+        ('patches/py3.8.1.patch', version_starts_with("3.10")),
     ]
 
     if shutil.which('lld') is not None:
-        patches = patches + [
+        patches += [
             ("patches/py3.7.1_fix_cortex_a8.patch", version_starts_with("3.7")),
             ("patches/py3.8.1_fix_cortex_a8.patch", version_starts_with("3.8")),
             ("patches/py3.8.1_fix_cortex_a8.patch", version_starts_with("3.9")),
-            ("patches/py3.8.1_fix_cortex_a8.patch", version_starts_with("3.10"))
+            ("patches/py3.8.1_fix_cortex_a8.patch", version_starts_with("3.10")),
+            ("patches/py3.8.1_fix_cortex_a8.patch", version_starts_with("3.11")),
         ]
 
+    # ctypes pythonapi fix.
+    # Taken from : https://github.com/gabomdq/ignifuga/blob/master/tools/patches/python.android.diff
+    patches += ["patches/ctypes-pythonapi.patch"]
+
```

Comment:
Since this issue historically affected all the previous versions, let's keep it separate, and let's target it into a separate PR.

So:
- The issue can be closed with a proper and specific PR
- We will have that announced in release notes.
- We can later (better) track that some work has been done.